packages/editor-ui/src/components/MainHeader/MainHeader.vue

```diff
@@ -27,181 +11,185 @@ import {
-			];
-		},
-		activeNode(): INodeUi | null {
-			return this.ndvStore.activeNode;
-		},
-		hideMenuBar(): boolean {
-			return Boolean(this.activeNode && this.activeNode.type !== STICKY_NODE_TYPE);
-		},
-		workflow(): IWorkflowDb {
-			return this.workflowsStore.workflow;
-		},
-		currentWorkflow(): string {
-			return String(this.$route.params.name || this.workflowsStore.workflowId);
-		},
-		onWorkflowPage(): boolean {
-			return !!(this.$route.meta.nodeView || this.$route.meta.keepWorkflowAlive);
-		},
-		readOnly(): boolean {
-			return this.sourceControlStore.preferences.branchReadOnly;
-		},
-	},
-	watch: {
-		$route(to, from) {
-			this.syncTabsWithRoute(to, from);
-		},
-	},
-	beforeMount() {
-		this.pushConnection.initialize();
-	},
-	mounted() {
-		this.dirtyState = this.uiStore.stateIsDirty;
-		this.syncTabsWithRoute(this.$route);
-	},
-	beforeUnmount() {
-		this.pushConnection.terminate();
-	},
-	methods: {
-		syncTabsWithRoute(to: RouteLocation, from?: RouteLocation): void {
-			if (
-				to.name === VIEWS.EXECUTION_HOME ||
-				to.name === VIEWS.WORKFLOW_EXECUTIONS ||
-				to.name === VIEWS.EXECUTION_PREVIEW
-			) {
-				this.activeHeaderTab = MAIN_HEADER_TABS.EXECUTIONS;
-			} else if (
-				to.name === VIEWS.WORKFLOW ||
-				to.name === VIEWS.NEW_WORKFLOW ||
-				to.name === VIEWS.EXECUTION_DEBUG
-			) {
-				this.activeHeaderTab = MAIN_HEADER_TABS.WORKFLOW;
-			}
+const router = useRouter();
+const route = useRoute();
+const locale = useI18n();
+const pushConnection = usePushConnection({ router });
+const ndvStore = useNDVStore();
+const uiStore = useUIStore();
+const sourceControlStore = useSourceControlStore();
+const workflowsStore = useWorkflowsStore();
+const executionsStore = useExecutionsStore();
 
-			if (to.params.name !== 'new' && typeof to.params.name === 'string') {
-				this.workflowToReturnTo = to.params.name;
-			}
+const activeHeaderTab = ref(MAIN_HEADER_TABS.WORKFLOW);
+const workflowToReturnTo = ref('');
+const executionToReturnTo = ref('');
+const dirtyState = ref(false);
 
-			if (
-				from?.name === VIEWS.EXECUTION_PREVIEW &&
-				to.params.name === from.params.name &&
-				typeof from.params.executionId === 'string'
-			) {
-				this.executionToReturnTo = from.params.executionId;
-			}
-		},
-		onTabSelected(tab: MAIN_HEADER_TABS, event: MouseEvent) {
-			const openInNewTab = event.ctrlKey || event.metaKey;
+const tabBarItems = computed(() => [
+	{ value: MAIN_HEADER_TABS.WORKFLOW, label: locale.baseText('generic.editor') },
+	{ value: MAIN_HEADER_TABS.EXECUTIONS, label: locale.baseText('generic.executions') },
+]);
 
-			switch (tab) {
-				case MAIN_HEADER_TABS.WORKFLOW:
-					void this.navigateToWorkflowView(openInNewTab);
-					break;
+const activeNode = computed(() => ndvStore.activeNode);
+const hideMenuBar = computed(() =>
+	Boolean(activeNode.value && activeNode.value.type !== STICKY_NODE_TYPE),
+);
+const workflow = computed(() => workflowsStore.workflow);
+const currentWorkflow = computed(() =>
```

Comment:
nit: Can you name this `workflowId` for clarity?

Reply:
done